### ISSUE TITLE
allowIndentation docs are outdated

### ISSUE THREAD

**Issue by RomanHotsiy** [markdoc/markdoc#469]
### What happened?

The [docs on the website](https://markdoc.dev/docs/faq#indentation) mention `allowIndentation` setting but it looks this setting is not functional: indentation is allowed regardless.

Should we adjust the docs?

### To reproduce

Use `allowIndentation: true` or `allowIndentation: false`, nothing changes.

### Version

0.3.4

### Additional context

Looks like it was disabled in this PR: https://github.com/markdoc/markdoc/pull/105

And here is the comment by @rpaul-stripe that explains the decision: https://github.com/markdoc/markdoc/issues/82#issuecomment-1165784291

**Comment by rpaul-stripe** [comment]
Yes, indent-based code blocks are now disabled by default. The `allowIndentation` option is still relevant, however, because it tells the tokenizer to allow other block types to be parsed when indented.

For example, consider the following Markdoc source:

```
This is a test

    - This is another test
```

When the `allowIndentation` option is not enabled, the list is not recognized as such because it is indented more than three spaces, resulting in this output: `<article><p>This is a test</p><p>- This is another test</p></article>`. With the option enabled, the list tag parses as expected.

The reason we still don't turn this on by default is because content written with this sort of indentation will not parse as expected in a CommonMark-compliant parser. Users who want their Markdoc content to still be portable to other parsers should probably keep it turned off.

I can update the docs to clarify this and address that indent code blocks are now off by default regardless of the setting. Thanks for raising this. 👍

**Comment by mDuo13** [comment]
Indented code blocks are a significant use case, though, especially with regards to portable content. Indented blocks are the only one supported by more "old-school" parsers including the original Gruber one, especially when dealing with content nested inside lists. For example, in Python-Markdown, even with the "extra" settings for code fences,  you can't use code fences in lists, but you can use indented code blocks. Content that's written for these parsers can also be handled by CommonMark parsers, but disabling indented code breaks that.

I say this as someone who has been writing such content for 9 years, and finds a lot of the content incompatible with Markdoc because of the use of indented code blocks inside lists.

**Comment by rpaul-stripe** [comment]
@mDuo13 if you really need to parse content written with indented code blocks in Markdoc, you can still subclass Markdoc's `Tokenizer` class and set up an instance of `MarkdownIt` with whatever features (and specific plugins) you want.

**Comment by maxblank-stripe** [comment]
<img width="800" alt="image" src="https://github.com/user-attachments/assets/15e11153-f043-42f0-b39d-ced1c8c755f1" />
